src/proto/message_translation/tbots_protobuf.h

```diff
@@ -12,11 +12,13 @@
  * Returns a TbotsProto::World proto given a World.
  *
  * @param world The world msg to extract the TbotsProto::World from
+ * @param sequence_number An optional sequence number for tracking the TbotsProto::World
  *
  * @return The unique_ptr to a TbotsProto::World proto containing the field, friendly
  * team, enemy team, ball, and the game state.
  */
-std::unique_ptr<TbotsProto::World> createWorld(const World& world);
+std::unique_ptr<TbotsProto::World> createWorld(const World& world,
+                                               const uint64_t sequence_number = -1);
```

Comment:
I think it would be more lightweight to have the sequence number just for the protobuf, because unlike the other parameters in the `World` C++ implementation, the sequence number is relevant for the protobuf only. Adding a sequence number to the `World` class would probably require a lot more modifications to the codebase (including the conversion tests you mentioned), which probably isn't necessary. 

I had intended for the `-1` to be a default value for the `sequence_number` in the `createWorld` function, but thanks for reminding me that `-1` isn't actually an unsigned int. One idea I have is to use `0` as the default value, and start the actual sequence numbers from `1`. How does that sound?

If we proceed with this idea, it would probably be helpful to change the sequence number of primitive sets to also start from `1`.